dynatrace/src/test/java/com/dynatrace/opentelemetry/metric/DynatraceMetricExporterTest.java

```diff
@@ -150,26 +665,51 @@ public void addDefaultDimensions() throws IOException {
         DynatraceMetricExporter.builder()
             .setApiToken("mytoken")
             .setUrl(connection.getURL())
-            .setDefaultDimensions(Labels.of("default", "value"))
             .build();
 
-    CompletableResultCode result = metricExporter.export(Collections.singleton(md), connection);
+    InMemoryDtMetricReader reader = new InMemoryDtMetricReader(metricExporter);
+    SdkMeterProvider sdkMeterProvider =
+        SdkMeterProvider.builder()
+            .setClock(testClock)
+            .registerMetricReader(reader)
+            .registerView(
+                InstrumentSelector.builder()
+                    .setInstrumentType(InstrumentType.COUNTER)
+                    .setInstrumentName("testSum")
+                    .build(),
+                View.builder()
+                    .setAggregation(Aggregation.sum(AggregationTemporality.CUMULATIVE))
+                    .build())
+            .build();
 
-    verify(connection).setRequestMethod("POST");
-    verify(connection).setRequestProperty("Authorization", "Api-Token mytoken");
-    verify(connection).setRequestProperty("Content-Type", "text/plain; charset=utf-8");
+    Meter sdkMeter = sdkMeterProvider.get(getClass().getName());
+    LongCounter counter = sdkMeter.counterBuilder("testSum").build();
+
+    counter.add(100);
+    sdkMeterProvider.forceFlush();
+    testClock.advance(Duration.ofSeconds(1));
+
+    counter.add(200);
+    sdkMeterProvider.forceFlush();
+    testClock.advance(Duration.ofSeconds(1));
+
+    counter.add(300);
+    testClock.advance(Duration.ofSeconds(1));
+    CompletableResultCode result = sdkMeterProvider.forceFlush();
+
+    assertTrue(result.isSuccess());
     assertEquals(
-        "name,default=value,dt.metrics.source=opentelemetry count,delta=194.0 1619687659000",
+        "testSum,dt.metrics.source=opentelemetry count,delta=200 1557212401000testSum,dt.metrics.source=opentelemetry count,delta=300 1557212403000",
```

Comment:
Why are these concatenated with a comma, and not with a newline?

Reply:
It was because the same instance of `ByteArrayOutputStream` was used across the flushes. I refactored it now so it is reset after each call to `flush`. Now the lines are asserted individually.